W3ChampionsStatisticService/WebApi/Authorization/BasicAuthConfiguration.cs

```diff
@@ -0,0 +1,48 @@
+using AspNetCore.Authentication.Basic;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+
+namespace W3ChampionsStatisticService.WebApi.Authorization
+{
+    public static class BasicAuthConfiguration
+    {
+        public const string ReadMetricsPolicy = "ReadMetrics";
+
+        public static IServiceCollection AddBasicAuthForMetrics(this IServiceCollection services)
+        {
+            // Get credentials from environment variables or fallback to defaults
+            string username = Environment.GetEnvironmentVariable("METRICS_ENDPOINT_AUTH_USERNAME") ?? "admin";
+            string password = Environment.GetEnvironmentVariable("METRICS_ENDPOINT_AUTH_PASSWORD") ?? "admin";
+
+            services.AddAuthentication(BasicDefaults.AuthenticationScheme)
+                .AddBasic(options =>
+                {
+                    options.Realm = "W3Champions Metrics";
+                    options.Events = new BasicEvents
+                    {
+                        OnValidateCredentials = async (context) =>
```

Comment:
There is no asynchronous code here so it should not have the async attribute

Reply:
Removed